proteus/test/test_gauges.py

```diff
@@ -0,0 +1,131 @@
+from math import ceil, sqrt, pow
+
+import numpy as np
+import numpy.testing as npt
+
+from proteus import (Comm,
+                     Domain,
+                     TransportCoefficients,
+                     MeshTools,
+                     FemTools,
+                     Transport,
+                     Quadrature,
+                     NumericalFlux)
+from proteus import default_p as p
+from proteus import default_n as n
+
+from proteus.Gauges import PointGauges
+
+from proteus.test.util import setup_profiling
+from nose.tools import eq_
+
+def build1DMesh(p, nnx):
+    return MeshTools.MultilevelEdgeMesh(nnx, 1, 1,
+                                        p.domain.L[0], 1, 1,
+                                        refinementLevels=1,
+                                        nLayersOfOverlap=0,
+                                        parallelPartitioningType=MeshTools.MeshParallelPartitioningTypes.node)
+def build2DMesh(p, nnx, nny):
+    return MeshTools.MultilevelTriangularMesh(nnx,nny,1,
+                                              p.domain.L[0], p.domain.L[1],1,
+                                              refinementLevels=1,
+                                              nLayersOfOverlap=0,
+                                              parallelPartitioningType=MeshTools.MeshParallelPartitioningTypes.node)
+
+def build3DMesh(p, nnx, nny, nnz):
+    return MeshTools.MultilevelTetrahedralMesh(nnx,nny,nnz,
+                                               p.domain.L[0], p.domain.L[1], p.domain.L[2],
+                                               refinementLevels=1,
+                                               nLayersOfOverlap=0,
+                                               parallelPartitioningType=MeshTools.MeshParallelPartitioningTypes.node)
+
+def gauge_setup():
+    comm = Comm.get()
+
+    #Simplified Physics
+    p.name="test_gauges"
+    p.nd = 3
+
+    class LinearSolution:
+        def uOfXT(self,x,t):
+            return (x[0]+10*x[1]+100*x[2])*(t+1.0)
+
+    p.initialConditions = {0:LinearSolution()}
+    p.dirichletConditions = {0: lambda x,flag: None}
+    p.domain = Domain.RectangularDomain(name="test_gauges_domain")
+    p.coefficients = TransportCoefficients.PoissonEquationCoefficients(aOfX = [lambda x: np.eye(p.nd, p.nd)],
```

Comment:
Since PoissonEquationCoefficients can generate a multi-component problem we could easily test the multi-component support of the gauges and set `p.coefficients.variableNames`  to something besides the default names `[`u0,u1,...]`.